tests/functional/triggering/01-or-conditional/flow.cylc

```diff
@@ -4,10 +4,10 @@
 [scheduling]
     [[graph]]
         R1 = """
-            a | b => c & d
-            c | d => e
-            b:failed => !b
-            c:failed => !c
+            a | b? => c? & d
+            c? | d => e
+            b:failed? => !b
+            c:failed? => !c
```

Comment:
Yes, but many of these tests pre-date spawn-on-demand, let alone optional outputs, and I've tended to adapt them minimally because it shows that Cylc 7 style use of suicide triggers still works even though it's not needed anymore (which is required for backward compatibility, of course).  We should make a sweep through all of these tests and fully update them, and then add a small number of well thought-out suicide trigger tests.

Reply:
#4374